NotifyMe/services/service.py

```diff
@@ -0,0 +1,202 @@
+                raise ValidationError("The 'subscription_plan' field is required.") # User: User object (31)
+
+            user = User.objects.create(subscription_plan=subscription_plan, **validated_data)
+            
+
+            # Create a new Subscription
+            start_date = timezone.now()
+            end_date = self.get_endtime(subscription_plan
+                                        , start_date)
+            if end_date is None:
+                raise ValidationError("Invalid subscription plan")
+
+            Subscription.objects.create(
+                user_id=user,
+                subscription_plan=subscription_plan,
+                start_date=start_date,
+                end_date=end_date    
+            )
+
+            logger.info(f"User created successfully with ID: {user.id}")
+
+            return user
+        except IntegrityError as e:
+            error_message = str(e)
+            logger.error(f"IntegrityError: {error_message}")
+            raise ValidationError(f"IntegrityError: {error_message}")
+        except Exception as e:
+            error_message = str(e)
+            logger.error(f"Unexpected error: {error_message}")
+            raise ValidationError(f"An error occurred: {error_message}")
+        
+        
+
+class SubscriptionService:
+    def get_subscription_data(self, request):
+        try:
+            subscriptions = Subscription.objects.all()
+            return subscriptions
+        except ObjectDoesNotExist as e:
+            logger.error(f"Object does not exist error: {e}")
+            return None
+        except MultipleObjectsReturned as e:
+            logger.error(f"Multiple objects returned error: {e}")
+            return None
+        except DatabaseError as e:
+            logger.error(f"Database error: {e}")
+            return None
+        except Exception as e:
+            logger.error(f"General error: {e}")
+            return None
+        
+    def get_subscriptionId_data(self, data):
+        subscription = Subscription.objects.get(id=data.get('id'))
+        return subscription
+    
+    
+    def get_subscriptionDatabase(self): # returns dataabse to check is the database exists
+        return Subscription
+    
+    
+    
+
+class NotificationService:
+    def get_notification_data(self, request):
+        try:
+            notifications = Notification.objects.all()
+            return notifications
+        except ObjectDoesNotExist as e:
+            logger.error(f"Object does not exist error: {e}")
+            return None
+        except MultipleObjectsReturned as e:
+            logger.error(f"Multiple objects returned error: {e}")
+            return None
+        except DatabaseError as e:
+            logger.error(f"Database error: {e}")
+            return None
+        except Exception as e:
+            logger.error(f"General error: {e}")
+            return None
+
+    def get_notification_type_data(self, request):
```

Comment:
this is constant you can create dict in a constant.py to get all the notification instead of fetching from db